gtsam/geometry/Rot2.h

```diff
@@ -50,6 +50,9 @@ namespace gtsam {
 
     /** default constructor, zero rotation */
     Rot2() : c_(1.0), s_(0.0) {}
+    
+    /** copy constructor */
+    Rot2(const Rot2& r) : Rot2(r.c_, r.s_) {}
```

Comment:
Is this not default?

Reply:
Do you mean is this not the copy constructor that is generated by default by the compiler? I imagine that it probably is, but it is hard to say whether it is the compiler-generated version of the copy constructor itself or the copy constructor elision for return value optimization that was causing the error without fully dissecting that assembly I posted in the issue reply.